src/Sulu/Component/CustomUrl/Manager/CustomUrlManager.php

```diff
@@ -0,0 +1,158 @@
+{
+    /**
+     * @var DocumentManagerInterface
+     */
+    private $documentManager;
+
+    /**
+     * @var CustomUrlRepository
+     */
+    private $customUrlRepository;
+
+    /**
+     * @var PathBuilder
+     */
+    private $pathBuilder;
+
+    public function __construct(
+        DocumentManagerInterface $documentManager,
+        CustomUrlRepository $customUrlRepository,
+        PathBuilder $pathBuilder
+    ) {
+        $this->documentManager = $documentManager;
+        $this->customUrlRepository = $customUrlRepository;
+        $this->pathBuilder = $pathBuilder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create($webspaceKey, array $data, $locale = null)
+    {
+        $document = $this->documentManager->create('custom_urls');
+        $this->bind($document, $data, $locale);
+
+        $this->documentManager->persist(
+            $document,
+            $locale,
+            [
+                'parent_path' => $this->getItemsPath($webspaceKey),
+                'node_name' => Urlizer::urlize($document->getTitle()),
+                'load_ghost_content' => true,
+            ]
+        );
+
+        return $document;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function readList($webspaceKey)
+    {
+        // TODO pagination
+
+        return $this->customUrlRepository->findList($this->getItemsPath($webspaceKey));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function read($uuid, $locale = null)
+    {
+        return $this->documentManager->find($uuid, $locale, ['load_ghost_content' => true]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function update($uuid, array $data, $locale = null)
+    {
+        $document = $this->read($uuid, $locale);
+        $this->bind($document, $data, $locale);
+
+        $this->documentManager->persist($document, $locale);
+
+        return $document;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete($uuid)
+    {
+        $this->documentManager->remove($this->read($uuid));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFields()
+    {
+        return [
+            'title' => ['property' => 'title'],
+            'published' => ['property' => 'published'],
+            'baseDomain' => ['property' => 'baseDomain'],
+            'domainParts' => ['property' => 'domainParts', 'type' => 'json_array'],
+            'target' => ['property' => 'target', 'type' => 'reference'],
+            'multilingual' => ['property' => 'multilingual'],
+            'canonical' => ['property' => 'canonical'],
+            'redirect' => ['property' => 'redirect'],
+            'targetLocale' => ['property' => 'targetLocale'],
+        ];
+    }
+
+    /**
+     * Bind data array to given document.
+     *
+     * @param CustomUrlDocument $document
+     * @param array $data
+     * @param string $locale
+     */
+    private function bind(CustomUrlDocument $document, $data, $locale)
```

Comment:
Its only a matter of creating a form .. and its less overhead than
leaving it here imo. Lets do things properly now rather than later,
otherwise we are just creating debt..

On Mon, Jan 18, 2016 at 05:51:59AM -0800, Wachter Johannes wrote:
>    In [1]src/Sulu/Component/CustomUrl/Manager/CustomUrlManager.php:
> 
>  > +            'target' => ['property' => 'target', 'type' => 'reference'],
>  > +            'multilingual' => ['property' => 'multilingual'],
>  > +            'canonical' => ['property' => 'canonical'],
>  > +            'redirect' => ['property' => 'redirect'],
>  > +            'targetLocale' => ['property' => 'targetLocale'],
>  > +        ];
>  > +    }
>  > +
>  > +    /**
>  > +     * Bind data array to given document.
>  > +     *
>  > +     * @param CustomUrlDocument $document
>  > +     * @param array $data
>  > +     * @param string $locale
>  > +     */
>  > +    private function bind(CustomUrlDocument $document, $data, $locale)
> 
>    we are currently not sure that we use forms for this mapping. because its
>    an overhead and we don't know if it worth it. we let this functionality in
>    this class for now and we think about a "global" solution later. /cc
>    [2]@chirimoya [3]@danrot
> 
>    —
>    Reply to this email directly or [4]view it on GitHub.
> 
>    Reverse link: [5]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu-io/sulu/pull/1927#discussion_r50000269
>    2. https://github.com/chirimoya
>    3. https://github.com/danrot
>    4. https://github.com/sulu-io/sulu/pull/1927/files#r50000269
>    5. https://github.com/sulu-io/sulu/pull/1927/files#r50000269